File: Лабораторна робота №2/TextLibrary/MyText.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using StringLibrary;

namespace TextLibrary
{
	public class MyText
    {
		public int count;
		public MyString[] arrgs = new MyString[100];
		public void AddStr_2(MyString text//Add the string
		{
			arrgs[count++] = text;
		}
		public void DeleteStr_2(int a)//Delete the string
		{
			for (int i = a - 1; i < count; i++)
			{
				arrgs[i] = arrgs[i + 1];
			}
			count--;
		}
		public void DeleteText_2()// Erase all the text
		{
			for (int i = 0; i < count; i++)
			{
				arrgs[i] = new MyString();
			}
			count = 0;
		}
		public void MinStr_2(char[] a)// Find min string
		{
			int min = arrgs[0].count;
			int i_m = 0;
			for (int i = 0; i < count; i++)
			{
				if (arrgs[i].count < min)
				{
					min = arrgs[i].count;
					i_m = i;
				}
			}
			int i_2 = 0;
			for (int j = 0; j < arrgs[i_m].count; j++)
			{
				a[i_2] = arrgs[i_m].arrgs[j];
				i_2++;
			}
		}
		public void FirstChars_2(char[] a)//Find string of fisrsts chars each sentencense
		{
			int j = 0;
			for (int i = 0; i < count; i++)
			{
				a[j] = arrgs[i].arrgs[0];
				j++;
			}
		}
		public int CharCountInTextAndStr_2(char c)// Find noumber one char in each sentencense
		{
			int s = 0;
			for (int i = 0; i < count; i++)
			{
				int Counte = arrgs[i].СharCount(arrgs[i], c);
				s += Counte;
			}
			return s;
		}
	}
}

File: Лабораторна робота №2/Лабораторна робота №2/MyText.cpp
#pragma once
#include "MyText.h"
#include "MyString.h"
#include <vector>
namespace Text
{
	void MyText::AddStr_2(MyString textForAdd)
	{
		arrgs.push_back(textForAdd);
	}
	void MyText::DeleteStr_2(int a)
	{
		arrgs.erase(arrgs.begin() + a - 1);
	}
	void MyText::DeleteText_2()
	{
		arrgs.clear();
	}
	void MyText::MinStr_2(std::vector<char>& a)
	{
		int min = arrgs[0].count;
		int i_m = 0;
		for (int i = 0; i < arrgs.size(); i++)
		{
			if (arrgs[i].count < min)
			{
				min = arrgs[i].count;
				i_m = i;
			}
		}
		for (int j = 0; j < arrgs[i_m].count; j++)
		{
			a.push_back(arrgs[i_m].arrgs[j]);
		}
	}
	void MyText::FirstChars_2(std::vector<char>& a)
	{
		for (int i = 0; i < arrgs.size(); i++)
		{
			a.push_back(arrgs[i].arrgs[0]);
		}
	}
	int MyText::CharCountInTextAndStr_2(char c)
	{
		int s = 0;
		for (int i = 0; i < arrgs.size(); i++)
		{
			int Counte = arrgs[i].ÑharCount(arrgs[i], c);
			s += Counte;
		}
		return s;
	}
}